File: java/rna-transcription/src/main/java/RnaTranscription.java
import java.util.Map;
import static java.util.Map.entry;
import java.util.stream.Collectors;

class RnaTranscription {

    private Map<String, String> dnaToRna = Map.ofEntries(entry("G", "C"), entry("C", "G"), entry("T", "A"), entry("A", "U"));

    String transcribe(String dnaStrand) {
        return dnaStrand.chars().mapToObj(nucleotide -> {
            return dnaToRna.get(Character.toString(nucleotide));
        }).collect(Collectors.joining(""));
    }

}

File: javascript/rna-transcription/rna-transcription.js
const DNA_TO_RNA = {
  G: 'C',
  C: 'G',
  T: 'A',
  A: 'U'
}

export const toRna = (strand) => {
  return strand
    .split('')
    .map(nucleotide => 
      DNA_TO_RNA[nucleotide])
    .join('');
};
